sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2102,6 +2102,16 @@ class Dataset[T] private[sql](
     Limit(Literal(n), logicalPlan)
   }
 
+  /**
+   * Returns a new Dataset by skipping the first `m` rows.
+   *
+   * @group typedrel
+   * @since 3.4.0
+   */
+  def offset(n: Int): Dataset[T] = withTypedPlan {
```

Comment:
please open a separate PR for it. This is adding a new user-facing API and needs more visibility.

Reply:
OK